broker.go

```diff
@@ -385,41 +382,49 @@ func (b *Broker) Deprovision(ctx context.Context, instanceID string, details bro
 	// Create the spec to return
 	var spec brokerapi.DeprovisionServiceSpec
 
+	b.instancesLock.Lock()
+	info, ok := b.instances[instanceID]
+	b.instancesLock.Unlock()
```

Comment:
Should we hold this lock until after the delete call? We reacquire the lock on ~426. Alternatively if `instancesLock` were to be changed to a `RWMutex` you could use a `RLock()`, but maybe that's overkill

Reply:
Sure, I could just hold the lock until the end of the method. It'll reduce code complexity.

Thanks for taking a look at this!